apps/studio/src/pages/sites/[siteId]/admin.tsx

```diff
@@ -0,0 +1,228 @@
+      await trpcUtils.site.getFooter.invalidate({ id: siteId })
+      toast({
+        title: "Saved site config!",
+        description: "Check your site in 5-10 minutes to view it live.",
+        status: "success",
+        ...BRIEF_TOAST_SETTINGS,
+      })
+    },
+    onError: () => {
+      toast({
+        title: "Error saving site config!",
+        description:
+          "If this persists, please report this issue at support@isomer.gov.sg",
+        status: "error",
+        ...BRIEF_TOAST_SETTINGS,
+      })
+    },
+  })
+
+  const [previousConfig] = trpc.site.getConfig.useSuspenseQuery({
+    id: siteId,
+  })
+  const [previousTheme] = trpc.site.getTheme.useSuspenseQuery({
+    id: siteId,
+  })
+  const [previousNavbar] = trpc.site.getNavbar.useSuspenseQuery({
+    id: siteId,
+  })
+  const [previousFooter] = trpc.site.getFooter.useSuspenseQuery({
+    id: siteId,
+  })
+
+  // NOTE: Refining the setNotificationSchema here instead of in site.ts since omit does not work after refine
+  const {
+    register,
+    handleSubmit,
+    formState: { isDirty, errors },
+  } = useZodForm({
+    schema: setSiteConfigAdminSchema
+      .omit({ siteId: true })
+      .refine((data) => !!data.config, {
+        message: "Site config must be present",
+        path: ["config"],
+      })
+      .refine((data) => !!data.theme, {
+        message: "Site theme must be present",
+        path: ["theme"],
+      })
+      .refine((data) => !!data.navbar, {
+        message: "Site navbar must be present",
+        path: ["navbar"],
+      })
+      .refine((data) => !!data.footer, {
+        message: "Site footer must be present",
+        path: ["footer"],
+      }),
+    defaultValues: {
+      config: JSON.stringify(previousConfig, null, 2),
+      theme: JSON.stringify(previousTheme, null, 2),
+      navbar: JSON.stringify(previousNavbar.content, null, 2),
+      footer: JSON.stringify(previousFooter.content, null, 2),
+    },
+  })
+
+  const [nextUrl, setNextUrl] = useState("")
+  const isOpen = !!nextUrl
+
+  useEffect(() => {
+    const handleRouteChange = (url: string) => {
+      if (isDirty) {
+        router.events.off("routeChangeStart", handleRouteChange)
+        setNextUrl(url)
+        router.events.emit("routeChangeError")
+        // eslint-disable-next-line @typescript-eslint/only-throw-error
+        throw "Error to abort router route change. Ignore this!"
+      }
+    }
+
+    if (!isOpen) {
+      router.events.on("routeChangeStart", handleRouteChange)
+    }
+    return () => {
+      router.events.off("routeChangeStart", handleRouteChange)
+    }
+  }, [isOpen, router.events, isDirty])
+
+  const onClickUpdate = handleSubmit((input) => {
+    mutate({
+      siteId,
+      ...input,
+    })
+  })
+
+  if (!isUserIsomerAdmin) {
+    return (
+      <Center h="full">
+        <Text>You do not have permission to access this page.</Text>
+      </Center>
+    )
+  }
```

Comment:
minor super nit: we should redirect + toast unauth users!

Reply:
Switched in d549351e72d5802c0e65a91969627e67cbb0d080.